Dockerfile

```diff
@@ -31,35 +31,51 @@ RUN pip install --target=/dependencies playwright~=1.27.1 \
 # Final image stage
 FROM python:3.10-slim
 
-RUN apt-get update && apt-get install -y --no-install-recommends \
-    libssl1.1 \
-    libxslt1.1 \
-    # For pdftohtml
-    poppler-utils \
-    zlib1g \
-    && apt-get clean && rm -rf /var/lib/apt/lists/*
-
-
-# https://stackoverflow.com/questions/58701233/docker-logs-erroneously-appears-empty-until-container-stops
-ENV PYTHONUNBUFFERED=1
 
-RUN [ ! -d "/datastore" ] && mkdir /datastore
+RUN set -ex; \
+    apt-get update && apt-get install -y --no-install-recommends \
+        gosu \
+        libssl1.1 \
+        libxslt1.1 \
+        # For pdftohtml
+        poppler-utils \
+        zlib1g && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*; \
+    useradd -u 911 -U -M -s /bin/false changedetection && \
+    usermod -G users changedetection; \
+    mkdir -p /datastore
 
 # Re #80, sets SECLEVEL=1 in openssl.conf to allow monitoring sites with weak/old cipher suites
 RUN sed -i 's/^CipherString = .*/CipherString = DEFAULT@SECLEVEL=1/' /etc/ssl/openssl.cnf
 
 # Copy modules over to the final image and add their dir to PYTHONPATH
 COPY --from=builder /dependencies /usr/local
-ENV PYTHONPATH=/usr/local
+ENV PYTHONPATH=/usr/local \
+    # https://stackoverflow.com/questions/58701233/docker-logs-erroneously-appears-empty-until-container-stops
+    PYTHONUNBUFFERED=1 \
+    # https://stackoverflow.com/questions/64808915/should-pycache-folders-be-included-in-production-containers
+    # This avoids permission denied errors because the app directory is root-owned.
+    PYTHONDONTWRITEBYTECODE=1 \
```

Comment:
isnt this going to slow down the app? or not?

re pytest cache, this is maybe better (set it to `datadir` ) https://docs.pytest.org/en/latest/reference/reference.html#confval-cache_dir

Reply:
> The advantage of cached bytecode is in startup time, not execution speed; import takes longer, but actual execution is just as fast (the bytecode is still there in memory, just not on disk). But in the case of containers, making the container larger means it takes longer to cache and launch, so you're wasting disk space and any gains in Python launch time are typically offset by delays pulling and launching the container in the first place. And writing out bytecode at runtime is pointless; it'll be lost when the container shuts down, so it's adding expense to startup for no benefit.

Reply:
ok thanks, yeah that makes sense